src/modules/jcms_rest/src/Plugin/Field/FieldTypeExtend/StringItem.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace Drupal\jcms_rest\Plugin\Field\FieldTypeExtend;
+
+use Drupal\Core\Field\FieldDefinitionInterface;
+use Drupal\Core\Field\Plugin\Field\FieldType\StringItem as StringItemExtend;
+
+/**
+ * {@inheritdoc}
+ */
+class StringItem extends StringItemExtend {
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function generateSampleValue(FieldDefinitionInterface $field_definition) {
+    switch ($field_definition->getName()) {
+      // Random values for orcid need to match the regex ^[0-9]{4}-[0-9]{4}-[0-9]{4}-[0-9]{3}[0-9X]$.
+      case 'field_person_orcid':
+        $orcid = [4, 4, 4, 3];
+        array_walk($orcid, function (&$value) {
+          $value = str_pad(rand(0, pow(10, $value-1)-1), $value, '0', STR_PAD_LEFT);
```

Comment:
if the intention is `rand(0, 9999)` shouldn't this be `rand(0, pow(10, $value)-1)`?

Reply:
Haha, that would explain why all my orcid's look like this:

```
0263-0960-0436-0243
0709-0135-0796-0678
0122-0915-0789-0534
0605-0983-0878-0911
0315-0248-0532-0196
0827-0698-0620-057X
0406-0695-0047-0184
0185-0333-0212-016X
0904-0019-0611-0614
0616-0510-0763-0644
0708-0073-0650-0430
0956-0685-0025-0054
0102-0335-0069-0452
0760-0024-0937-0380
0172-0457-0541-0165
0076-0079-0816-0469
0210-0268-0934-0498
0857-0649-0846-0815
0045-0960-0608-0754
0544-0656-0833-0435
0991-0551-0559-0097
0586-0965-0462-0969
0316-0073-0195-0987
0176-0541-0829-0913
0413-0737-0389-0379
```